packages/ui/src/components/search-files.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useCallback, useMemo, useState } from 'react'
+
+import {
+  Command,
+  CommandEmpty,
+  CommandGroup,
+  CommandItem,
+  CommandList,
+  Popover,
+  PopoverAnchor,
+  PopoverContent,
+  SearchBox,
+  Text
+} from '@/components'
+
+interface SearchFilesProps {
+  navigateToFile: (file: string) => void
+  filesList: string[] | undefined
+}
+
+export const SearchFiles = ({ navigateToFile, filesList }: SearchFilesProps) => {
+  const [isOpen, setIsOpen] = useState(false)
+  const [query, setQuery] = useState('')
+
+  const handleInputChange = useCallback((event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value
+    setQuery(value)
+    setIsOpen(value !== '')
+  }, [])
+
+  const filteredFiles = useMemo(() => {
+    return filesList?.filter(file => file.toLowerCase().includes(query.toLowerCase()))
+  }, [query, filesList])
```

Comment:
You can simplify things, by always making sure an array is returned. Then you can use without optional chaining below. Also, this will run whether or not a query has been entered, so you might as well return quickly in that case.

```suggestion
  const filteredFiles = useMemo<string[]>(() => {
    if (!filesList) return []
    if (!query) return filesList
    
    const queryText = query.toLowerCase()
  
    return filesList.filter(file => file.toLowerCase().includes(queryText))
  }, [query, filesList])
```